app/models/ext_management_system.rb

```diff
@@ -260,6 +264,78 @@ def default_endpoint
     default || endpoints.build(:role => "default")
   end
 
+  # TODO: takes multiple connection data
+  # endpoints, authentications, and works out what's what.
+  def connections=(options)
+    options.each do |option|
+      endpoint = endpoint_builder(option[:endpoint])
+      authentication_builder(option[:authentication])
+        # connection_by_role=(option)
+    end
+  end
+
+  def connections
```

Comment:
As a WIP, I have for example:
```ruby
config.endpoint.amqp_hostname
config.authentication.amqp_username
```
where `amqp_hostname` and `amqp_username` are generated from the UI field names.